library/core/src/slice/mod.rs

```diff
@@ -1167,6 +1167,66 @@ impl<T> [T] {
         (remainder, array_slice)
     }
 
+    /// Obtains a reference to the `N`-element array located at the range `idx..idx+N`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `N` is 0, or `idx + N` is out of bounds.
```

Comment:
that seems unnecessary to me, getting a `[T; 0]` chunk seems completely fine to me.

The issue with the chunks iterators is that they would never terminate for chunks of size 0.